src/configuration.ts

```diff
@@ -120,8 +120,80 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config = JSON.parse(fileData) as IConfigFile;
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames != null) {
+            const configFileDir = path.dirname(configFilePath);
+            if (typeof parentConfigNames === "string") {
+                parentConfigNames = [<string>parentConfigNames];
+            }
+
+            for (const name of parentConfigNames) {
+                const parentConfigFilePath = resolveConfigFilePath(name, configFileDir);
+
+                let parentConfig = loadConfigurationFromPath(parentConfigFilePath);
+                config = applyExtends(config, parentConfig);
+            }
+        }
+
+        return config;
+    }
+}
+
+export interface IConfigFile {
+    extends: string | string[];
+    rulesDirectory: string | string[];
+    rules: any;
+}
+
+/**
+ * Resolve config file path.
+ * @var {string} relativeFilePath Relative path or
+ *   package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigFilePath(relativeFilePath: string, relativeTo: string) {
+    // `relativeFilePath.indexOf("/") === -1` will fail sub-rule use case when it is supported.
+    if (relativeFilePath.indexOf("/") === -1 && relativeFilePath.indexOf("tslint-config") === -1) {
+        relativeFilePath = `tslint-config-${relativeFilePath}`;
     }
+
+    if (relativeFilePath.indexOf("tslint-config-") === 0) {
+        let fileData = fs.readFileSync(path.resolve(`node_modules/${relativeFilePath}/package.json`), "utf8");
```

Comment:
I see. make sense. However, what should the package expose?
```ts
import * as tslintJson form 'somepackage';
import tslintJson from 'somepackage';
```
or something else? Is there a fixed convention?
Because from `tslint-eslint-rules`, it asks consumer to:

```js
{
  "ruleDirectory": "node_modules/tslint-eslint-rules/dist/rules"
}
```
So I'm not sure what the convention should be.

Do you have defined this convention?